### PR TITLE
[signer] time out an empty sortition and mark a miner as misbehaving if an initial block proposal fails to arrive in time

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/stacks-network/stacks-core/issues/4993

### CODE REVIEW

File: stacks-signer/src/chainstate.rs [hunk 1]
```diff
@@ -147,12 +150,40 @@ impl<'a> ProposedBy<'a> {
...
+                    warn!(
+                        "Miner proposed first block after block proposal timeout.";
+                        "proposed_block_consensus_hash" => %block.header.consensus_hash,
+                        "proposed_block_signer_sighash" => %block.header.signer_signature_hash(),
+                        "current_sortition_consensus_hash" => ?self.cur_sortition.consensus_hash,
+                        "last_sortition_consensus_hash" => ?self.last_sortition.as_ref().map(|x| x.consensus_hash),
+                        "burn_block_received_time" => ?received_time,
+                    );
+                    self.cur_sortition.miner_status =
+                        SortitionMinerStatus::InvalidatedBeforeFirstBlock;
```
Comment (kantai) [inline stacks-signer/src/chainstate.rs:183]: I would add a `return false` here as well.

**Review by kantai — COMMENTED** [review]
This change looks good to me, I think it just needs some integration testing.

File: stacks-signer/src/chainstate.rs (hunk 1 above)
Comment (jferrant) [reply stacks-signer/src/chainstate.rs:183]: But isn't it possible that it is still valid? what if the proposal is delayed but builds off the prior sortition?

Comment (kantai) [reply stacks-signer/src/chainstate.rs:183]: Late arrival is enough to be considered "invalid" -- this discourages miners from delaying their proposals. If proposals arrive late, it increases the pressure on the network to get blocks confirmed before the next bitcoin block (and other miners who are competing in the next block would need to be able to get their block commits in time).

Comment (jferrant) [reply stacks-signer/src/chainstate.rs:183]: Then what is the point of setting this sortition state and then later checking if the block is build off the last sortition?

**Review by kantai — DISMISSED** [review]
This LGTM, but there are some merge conflicts which will have to be resolved.

**Review by jcnelson — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Pull request merged**